index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
-
-      /**
-       * Checks if a token has been added to ArConnect
-       * 
-       * @param id Token to check for
-       * 
-       * @returns Token added or not
-       */
-      isTokenAdded(id: string): Promise<boolean>;
-
-      /**
-       * Dispatch an Arweave transaction (preferably bundled)
-       *
-       * @param transaction Transaction to dispatch
-       * 
-       * @returns Dispatched transaction ID and type
-       */
-      dispatch(transaction: Transaction): Promise<DispatchResult>;
-
-      /**
-       * Create a deterministic secret based on the input data.
-       * 
-       * @param data Input data to generate the hash from
-       * @param options Hash configuration
-       * 
-       * @returns Hash array
-       */
-      privateHash(data: ArrayBuffer, options: SignMessageOptions): Promise<Uint8Array>;
-
-      /**
-       * Create and sign a DataItem (bundled transaction item),
-       * that can be loaded into "arbundles".
-       * 
-       * @param dataItem Data item params
-       * 
-       * @returns Buffer of the signed data item
-       */
-      signDataItem(dataItem: DataItem): Promise<ArrayBufferLike>;
-
-      /**
-       * Create a cryptographic signature for any piece of data for later validation.
-       * This function cannot be used to sign transactions or interactions, because the data
-       * gets hashed before the signature generation.
-       * 
-       * @param data Message to be hashed and signed
-       * @param options Signature options
-       * 
-       * @returns Signature
-       */
-      signMessage(data: ArrayBuffer, options?: SignMessageOptions): Promise<Uint8Array>;
-
-      /**
-       * Verify a cryptographic signature created with the arweaveWallet.signMessage() function.
-       * 
-       * @param data Data to verify with the signature
-       * @param signature Signature to validate
-       * @param publicKey Optionally match the signature with a different public key than the currently active
-       * @param options Signature options
-       * 
-       * @returns Validity
-       */
-      verifyMessage(
-        data: ArrayBuffer,
-        signature: ArrayBuffer | string,
-        publicKey?: string,
-        options?: SignMessageOptions
-      ): Promise<boolean>;
-
-      /**
-       * Experimental event emitter. Allows listening to gateway config
-       * updates, bundler node changes, etc.
-       */
-      events: Emitter<InjectedEvents>;
-    };
+    arweaveWallet: ArweaveWallet;
   }
+  
   interface WindowEventMap {
     walletSwitch: CustomEvent<{ address: string }>;
     arweaveWalletLoaded: CustomEvent<{}>;
   }
 }
 
+export type TypedArray =
+  | Int8Array
+  | Uint8Array
+  | Uint8ClampedArray
+  | Int16Array
+  | Uint16Array
+  | Int32Array
+  | Uint32Array
+  // | Float16Array
+  | Float32Array
+  | Float64Array
+  | BigInt64Array
+  | BigUint64Array;
+
+export type BinaryDataType = ArrayBuffer | TypedArray | DataView; // | Buffer;
```

Comment:
Various functions in the docs that take a binary data type accept multiple values, as you can see here:

![image](https://github.com/user-attachments/assets/033aae48-92b0-4cc1-94b7-b52c082f9239)

Depending on the implementation, `Buffer` and even `string` could be accepted. In Othent, only `decrypt` accepts `BinaryDataType` but not `string`.